microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
+The memory graph starts with a GC root, which may be the `window` object of the browser or the `Global` object of a Node.js module. You don't control how the root object is garbage-collected.
 
-The memory graph starts with a root, which may be the `window` object of the browser or the `Global` object of a Node.js module.  You don't control how the root object is garbage-collected.
+Nodes that aren't reachable from the root can get garbage-collected.
 
 ![You can't control how the root object is garbage-collected](./memory-101-images/dontcontrol.png)
 
-Whatever isn't reachable from the root gets garbage-collected.
-
-> [!NOTE]
-> The number that's shown in the [Shallow size](#shallow-size) and [Retained size](#retained-size) columns is the number of bytes.
-
 
 <!-- ====================================================================== -->
-## Objects retaining tree
+## Object sizes and distances
 
-The heap is a network of interconnected objects.  In the mathematical world, this structure is called a _graph_ or _memory graph_.  A graph is constructed from _nodes_ that are connected by _edges_.
+When working with the **Memory** tool, you will likely find yourself looking at the following columns of information:
 
-Nodes and edges in a graph are given labels as follows:
+* **Shallow Size**
+* **Retained Size**
+* **Distance**
 
-*  _Nodes_ (or _objects_) are labelled using the name of the _constructor_ function that was used to build them.
+The numbers that are shown in the **Shallow size** and **Retained size** columns are the number of bytes.
 
-*  _Edges_ are labelled using the names of _properties_.
-
-Learn [how to record a profile using the Heap Profiler](heap-snapshots.md).  In the following figure, some of the notable things in the Heap Snapshot recording in the [Memory](heap-snapshots.md) tool include **Distance**, which is the distance from the garbage-collection root.  If almost all the objects of the same type are at the same distance, and a few are at a bigger distance, that's something worth investigating.
-
-Distance from root:
+![Shallow and Retained Size](./memory-101-images/shallow-retained.png)
 
-![Distance from root](./memory-101-images/root.png)
+<!-- ------------------------------ -->
+#### Shallow size
 
+The _shallow size_ is the size of the memory that is held by the object directly.
 
-<!-- ====================================================================== -->
-## Dominators
+The shallow size of an object is usually small because JavaScript objects often just store their own descriptions only in memory. Arrays and strings are exceptions to this and can have a significant shallow size. Most JavaScript objects store their values elsewhere in the renderer memory, and only expose a small wrapper object on the JavaScript heap.
```

Comment:
> Most JavaScript objects store their values elsewhere in the renderer memory, and only expose a small wrapper object on the JavaScript heap.

By the way, I'm working on changing this so that shallow sizes will be more useful. The experimental setting will land upstream soon (probably tomorrow). Work is tracked [here](https://bugs.chromium.org/p/chromium/issues/detail?id=1517057).

Reply:
Cool, that sounds very useful. Once the new setting is in Canary, we should document it too.